wallet_presentation/src/commonMain/kotlin/net/thechance/mena/wallet/presentation/screen/statementsHistory/component/AnimatedStatementItem.kt

```diff
@@ -0,0 +1,136 @@
+package net.thechance.mena.wallet.presentation.screen.statementsHistory.component
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.core.LinearEasing
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.shrinkVertically
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.offset
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.TransformOrigin
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import net.thechance.mena.designsystem.presentation.theme.theme.Theme
+import net.thechance.mena.wallet.presentation.screen.statementsHistory.StatementsHistoryScreenState
+import org.jetbrains.compose.ui.tooling.preview.Preview
+
+@Composable
+fun AnimatedStatementItem(
+    statement: StatementsHistoryScreenState.StatementItem,
+    lastStatement: StatementsHistoryScreenState.StatementItem,
```

Comment:
We can simplify this by passing isDividerVisible or isLastItem flag instead of passing the whole last item for each item in the list.

```suggestion
    isDividerVisible: Boolean,
```